src/test/java/CustomListTest.java

```diff
@@ -0,0 +1,200 @@
+    }
+
+    @Test
+    void testSizeDefaultCapacity() {
+        for (int i = 0; i < 100; i++) {
+            customList.add(i);
+            arrayList.add(i);
+            Assertions.assertEquals(customList.size(), arrayList.size());
+        }
+
+        Assertions.assertEquals(customList.get(99), arrayList.get(99));
+    }
+
+    @Test
+    void testSizeCustomCapacity() {
+        customList = new CustomList(1);
+        for (int i = 0; i < 100; i++) {
+            customList.add(i);
+            arrayList.add(i);
+            Assertions.assertEquals(customList.size(), arrayList.size());
+        }
+
+        Assertions.assertEquals(customList.get(99), arrayList.get(99));
+    }
+
+    @Test
+    void testClear() {
+        customList.add("First");
+        customList.add("Second");
+        arrayList.add("First");
+        arrayList.add("Second");
+
+        customList.clear();
+        arrayList.clear();
+
+        Assertions.assertEquals(customList.size(), arrayList.size());
+        Assertions.assertTrue(customList.isEmpty());
+        Assertions.assertTrue(arrayList.isEmpty());
+    }
+
+    @Test
+    void testAddAndRemove() {
+        int size = 100;
+
+        for (int i = 0; i < size; i++) {
+            customList.add(i);
+            arrayList.add(i);
+        }
+
+        for (int i = size - 1; i >= 0; i--) {
+            customList.remove(i);
+            arrayList.remove(i);
+            Assertions.assertEquals(customList.size(), arrayList.size());
+        }
+    }
+
+    @Test
+    void testAddAndRandomRemove() {
+        int size = 100;
+
+        for (int i = 0; i < size; i++) {
+            customList.add(i);
+            arrayList.add(i);
+        }
+
+        for (int i = size - 1; i >= 0; i--) {
+            int randomNumber = (int) (Math.random() * (i + 1));
+
+            customList.remove(randomNumber);
+            arrayList.remove(randomNumber);
+        }
+
+        Assertions.assertEquals(customList.size(), arrayList.size());
+    }
+
+    @Test
+    void testNullValues() {
+        customList.add(null);
+        customList.add("NotNull");
+        customList.add(null);
+
+        arrayList.add(null);
+        arrayList.add("NotNull");
+        arrayList.add(null);
+
+        Assertions.assertEquals(customList.size(), arrayList.size());
+        Assertions.assertEquals(customList.get(0), arrayList.get(0));
+        Assertions.assertEquals(customList.get(2), arrayList.get(2));
+        Assertions.assertNull(customList.get(0));
+        Assertions.assertNull(arrayList.get(0));
+    }
+
+    @Test
+    void testEmptyListIndex() {
+        Assertions.assertTrue(customList.isEmpty());
+        Assertions.assertTrue(arrayList.isEmpty());
+        Assertions.assertEquals(customList.size(), arrayList.size());
+
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> customList.get(0));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> arrayList.get(0));
+    }
+}
```

Comment:
try to use @ParametrizedTest to avoid code duplication 
we'll need this in the future as well